Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -257,21 +254,16 @@ final class ConfiguringPipelineAsyncMultiplexerTests: XCTestCase {
         // Let's pretend the TLS handler did protocol negotiation for us
         self.serverChannel.pipeline.fireUserInboundEventTriggered(TLSUserEvent.handshakeCompleted(negotiatedProtocol: "h2"))
 
-        let nioProtocolNegotiationResult = try await nioProtocolNegotiationHandler.protocolNegotiationResult.get()
+        let negotiationResult = try await nioProtocolNegotiationResult.get().waitForFinalResult()
 
         try await assertNoThrow(try await self.assertDoHandshake(client: self.clientChannel, server: self.serverChannel))
 
         let serverMultiplexer: NIOHTTP2Handler.AsyncStreamMultiplexer<Channel>
-        switch nioProtocolNegotiationResult {
-        case .deferredResult:
+        switch negotiationResult {
+        case .http1_1:
             preconditionFailure("Negotiation result must be ready")
```

Comment:
```suggestion
            preconditionFailure("Negotiation result must be HTTP/2")
```
I was going to fix this in a future PR but we may as well do it now